src/Test/Perf/Runner/Program.cs

```diff
@@ -16,23 +16,76 @@ namespace Runner
 {
     public static class Program
     {
-        public static void Main(string[] args)
+        public static int Main(string[] args)
         {
 
             bool shouldReportBenchview = false;
             bool shouldUploadTrace = true;
             bool isCiTest = false;
+            string submissionName = null;
+            string submissionType = null;
             string traceDestination = @"\\mlangfs1\public\basoundr\PerfTraces";
+            string branch = null;
 
             var parameterOptions = new OptionSet()
             {
-                {"report-benchview", "report the performance retults to benview.", _ => shouldReportBenchview = true},
+                {"report-benchview", "report the performance results to benchview.", _ => shouldReportBenchview = true},
+                {"benchview-submission-type=", $"submission type to use when uploading to benchview ({String.Join(",", ValidSubmissionTypes)})", type => { submissionType = type; } },
+                {"benchview-submission-name=", "submission name to use when uploading to benchview (required for private and local submissions)", name => { submissionName = name; } },
+                {"branch=", "name of the branch you are measuring on", name => { branch = name; } },
                 {"ci-test", "mention that we are running in the continuous integration lab", _ => isCiTest = true},
                 {"no-trace-upload", "disable the uploading of traces", _ => shouldUploadTrace = false},
                 {"trace-upload_destination", "set the trace uploading destination", loc => { traceDestination = loc; }}
             };
             parameterOptions.Parse(args);
 
+            if (shouldReportBenchview)
```

Comment:
I'd pull this out of the main method to make things easier to read top-to-bottom